### PR TITLE
Checkout: Block removal of Private Registration from cart only when renewing

### LINKED ISSUE #1410: Domains: Can't remove Private Registration upgrade from the cart
Steps to reproduce:
1. Go to http://calypso.localhost:3000/domains/manage.
2. Select a Site.
3. Select a Domain without Private Registration.
4. Navigate to Contacts and Privacy page.
5. Navigate to Privacy Protection page.
6. Click Add Privacy Protection button.
7. Checkout page will load and you won't be able to remove this upgrade from the cart (remove icon is missing):

screen shot 2015-12-08 at 11 14 19

### PR DESCRIPTION
**Pull request by gziolo**
Fixes #1410.

Steps to reproduce:
1. Go to http://calypso.localhost:3000/domains/manage.
2. Select a Site.
3. Select a Domain without Private Registration.
4. Navigate to Contacts and Privacy page.
5. Navigate to Privacy Protection page.
6. Click Add Privacy Protection button.
7. Checkout page will load and you won't be able to remove this upgrade from the cart (remove icon is missing):
### Testing
1. Do (1)-(6) from steps to reproduce.
2. You should be able to remove Private Registration from the cart.
3. Go to http://calypso.localhost:3000/purchases.
4. Select a Domain purchase with Private Registration that is expiring.
5. Click Renew button.
6. You shouldn't be able to remove Private Registration from the cart. Domain needs to be renewed together with Private Registration.

### CODE REVIEW

**Comment by gziolo** [comment]
@scruffian @fabianapsimoes: Can you confirm it doesn't break anything?

File: client/lib/cart-values/index.js [hunk 1]
```diff
@@ -24,6 +24,18 @@ function applyCoupon( coupon ) {
 	};
 }
 
+function canRemoveFromCart( cart, cartItem ) {
```
Comment (gziolo) [inline client/lib/cart-values/index.js]: This function fits much better in cart related logic.

**Comment by drewblaisdell** [comment]
This works properly and improves the quality of the code it touches. I tested both domain cases. LGTM.

**Comment by gziolo** [comment]
Thanks @drewblaisdell for review! :)

**Pull request merged**